content/en/blog/_posts/2022-04-18-volume-expansion-ga.md

```diff
@@ -0,0 +1,84 @@
+---
+layout: blog
+title: "Volume expansion going GA in v1.24"
+date: 2022-04-18
+slug: volume-expansion-ga
+---
+
+**Author:** Hemant Kumar
+
+## Introduction
+
+Volume expansion was introduced as a alpha feature in Kubernetes 1.8 and it went beta in 1.11 and with Kubernetes 1.24 we are excited to announce general availability(GA)
+of volume expansion feature.
+
+This feature allows Kubernetes users to simply edit their `PersistentVolumeClaim` objects and specify new size in PVC Spec and Kubernetes will automatically expand the volume
+using storage backend and also expand the underlying file system in-use by the Pod without requiring any downtime at all if possible.
+
+
+### How to use volume expansion feature
+
+Using volume expansion is as simple as editing `storage` field of pvc spec and specifying new size. For example - given following PVC:
+
+```
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: myclaim
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi --> specify new size here
```

Comment:
Saying "sizes are in bytes" is not entirely accurate because as an API field in JSON (at least the one user is editing) it is basically a *string*.  Sure it can be bytes but we also try and best to convert string values like `1Gi` or `10GB` to appropriate byte sizes internally. 

I am not sure if a blog article is right place to inform users about nuances associated with `resource.Quantity` type and how it is serialized internally.

Reply:
Serialization and any Golang implementation details aren't important, but folks reading this article might well not know that Kubernetes has implied units (`storage: 1Gi` not `storageBytes: 1Gi`, nor `storage: 1GiB`). I have definitely lost count of the number of production Kubernetes clusters I've seen where key workloads share nodes with Pods that don't have requests and limits...

We should aim to cover things for those less experienced readers. If people have to fire up a Google search to understand what they're reading here, I don't think we've succeeded.

You can link to https://kubernetes.io/docs/reference/glossary/?all=true#term-quantity if that helps.